### PR TITLE
Fix the wrong style of icon in Mozilla Firefox, use margin instead of transform to set the position

### PR DESCRIPTION
**Pull request by lixiang810**
#8 #9

In 6e5cafa, I used margin to replace translateY, but I think use flex would be better. However, some animate does not work well after that, so I reverted the change and wrote this to remind you to considerate about using flex to set the position of Toasts.

Tips: merge feat1 (#10) first.

### CODE REVIEW

**Comment by ardeora** [comment]
Hey there! This is an interesting approach I love it! I have some suggested changes that I will include in the next comment

File: src/components/LoaderIcon.tsx [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+  const animation = `${rotate} 0.75s linear infinite`;
   return (
-    <svg style={{overflow: 'visible'}} viewBox="0 0 32 32">
-      <path fill="none" stroke={props.primary || "#E5E7EB"} stroke-width="4" stroke-miterlimit="10" d="M16,6c3,0,5.7,1.3,7.5,3.4c1.5,1.8,2.5,4,2.5,6.6c0,5.5-4.5,10-10,10S6,21.6,6,16S10.5,6,16,6z"/>
-      <path style={{animation, 'transform-origin': '50% 50%'}}  fill="none" stroke={props.secondary || "#4b5563"} stroke-width="4" stroke-linecap="round" stroke-miterlimit="10" d="M16,6c3,0,5.7,1.3,7.5,3.4c0.6,0.7,1.1,1.4,1.5,2.2"/>
+    <svg
+      style={{ overflow: "visible" }}
+      viewBox="0 0 32 32"
+      width="2rem"
+      height="2rem"
```
Comment (ardeora) [inline src/components/LoaderIcon.tsx:11]:
```suggestion
      width="1.25rem"
      height="1.25rem"
```

Same here. I think `1.25rem` looks like a better default size for all toasts. `2rems` is too big. :D

File: src/components/SuccessIcon.tsx [hunk 2]
```diff
@@ -1,17 +1,50 @@
...
   const secondaryCircle = `${pingCircle} 1s cubic-bezier(0, 0, 0.2, 1) forwards`;
-  const check = `${icon} 0.2s ease-out forwards`
+  const check = `${icon} 0.2s ease-out forwards`;
 
   return (
-    <svg style={{overflow: 'visible'}} viewBox="0 0 32 32">
-      <circle style={{animation: mainCircle, 'transform-origin': '50% 50%', 'animation-delay': '100ms', opacity: 0}}  fill={props.primary || '#34C759'} cx="16" cy="16" r="16"/>
-      <circle style={{animation: secondaryCircle, 'transform-origin': '50% 50%', 'animation-delay': '250ms'}} fill={props.primary || '#34C759'} cx="16" cy="16" r="12"/>
-      <path style={{animation: check, 'stroke-dasharray': 22, 'stroke-dashoffset': 22, 'animation-delay': '250ms'}} fill="none" stroke={props.secondary || '#FCFCFC'} stroke-width="4" stroke-linecap="round" stroke-miterlimit="10" d="M9.8,17.2l3.8,3.6c0.1,0.1,0.3,0.1,0.4,0l9.6-9.7"/>
+    <svg style={{ overflow: "visible" }} viewBox="0 0 32 32" width="2rem" height="2rem">
```
Comment (ardeora) [inline src/components/SuccessIcon.tsx:10]:
```suggestion
    <svg style={{ overflow: "visible" }} viewBox="0 0 32 32" width="1.25rem" height="1.25rem">
```

I think `1.25rem` looks like a better default size for all toasts. `2rems` is too big. :D

File: vite.config.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { resolve } from "path";
+import { defineConfig } from "vite";
+import dts from "vite-plugin-dts";
+import solid from "vite-plugin-solid";
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [solid(), dts()],
+  root: resolve("./dev"),
+});
```
Comment (ardeora) [inline vite.config.ts:10]: Can we move this file inside the `dev` folder. This will help keep things cleaner in the root directory

**Review by ardeora — CHANGES_REQUESTED** [review]

**Comment by ardeora** [comment]
Also thanks for taking a look and making the suggested changes. This will make solid toast so much better 💯

**Comment by lixiang810** [comment]
Maybe I should make 1 pull request instead of 2, now we will have some merge conflicts. 😂

**Comment by ardeora** [comment]
Thanks again for these changes. I have published v0.2.0 with all your changes! Let me know if something doesn't work for you 🕺

**Comment by lixiang810** [comment]
It works very nice now, nice work. Thanks for reviewing my code. 👍

**Pull request merged**